src/domain/parties/parties.js

```diff
@@ -67,13 +67,31 @@ const getPartiesByTypeAndID = async (headers, params, method, query, span = unde
     const errorCallbackEndpointType = partySubIdOrType ? Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTIES_SUB_ID_PUT_ERROR : Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTIES_PUT_ERROR
     const requesterParticipantModel = await participant.validateParticipant(headers[Enums.Http.Headers.FSPIOP.SOURCE], childSpan)
     if (requesterParticipantModel) {
+      let options = {
+        partyIdType: type,
+        partyIdentifier: params.ID
+      }
+      options = partySubIdOrType ? { ...options, partySubIdOrType } : options
+
+      // see https://github.com/mojaloop/design-authority/issues/79
+      if (headers[Enums.Http.Headers.FSPIOP.DESTINATION]) {
+        // the requester has specifid a destination routing header. We should respect that and forward the request directly to the destination
+        // without consulting any oracles.
+        
+        // first check the destination is a valid participant
+        const destParticipantModel = await participant.validateParticipant(headers[Enums.Http.Headers.FSPIOP.DESTINATION], childSpan)
+        if(!destParticipantModel) {
+          Logger.error('Destination FSP not found')
+          throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Destination FSP not found')
+        }
+
+        // all ok, go ahead and forward the request
+        await participant.sendRequest(headers, headers[Enums.Http.Headers.FSPIOP.DESTINATION], callbackEndpointType, Enums.Http.RestMethods.GET, undefined, options, childSpan);
+        return;
```

Comment:
I believe you should be closing out the `childSpan` and `histTimerEnd`.

I suggest that you should remove the `return` and rather add an "else" to your if-statement, and let those be handled by the "finally" of the try-catch.

Or alternatively:

```suggestion
        histTimerEnd({ success: true })
        if (childSpan && !childSpan.isFinished) {
          childSpan.finish()
        }
        return;
```

Additionally, you have a `semi-colon` at the end which I believe should fail the lint rules.

Reply:
thanks. will fix.